bugz.py

```diff
@@ -0,0 +1,202 @@
+            description = entry[1]
+            severity = entry[3]
+            priority = entry[4]
+            status = entry[5]
+            resolution = entry[6]
+            # If there are additional fields (sub-entry), iterate over them
+            if entry[2]:
+                for sub_entry in entry[2]:
+                    row = {"bug_id": bug_id, "description": description,
+                           **sub_entry, "severity": severity,
+                           "priority": priority,
+                           "bug_status": status, "resolution": resolution}
+                    rows.append(row)
+            else:
+                # If no sub-entry, just add the bug_id description
+                row = {"bug_id": bug_id, "description": description,
+                       "severity": severity, "priority": priority,
+                       "priority": priority, "bug_status": status,
+                       "resolution": resolution}
+                rows.append(row)
+
+        # Create the DataFrame
+        df = pd.DataFrame(rows)
+
+        df['modification_date'] = pd.to_datetime(df['modification_date'], format='%Y%m%dT%H:%M:%S') # noqa
+        df['creation_date'] = pd.to_datetime(df['creation_date'], format='%Y%m%dT%H:%M:%S') # noqa
+
+        # Drop the columns 'type_id' and 'id'
+        df_cleaned = df.drop(columns=["type_id", "id"])
+
+        data_frame = self.db.report_bugzilla_qa_needed(df_cleaned)
+        self.db.report_bugzilla_qa_needed_insert(data_frame)
+
+        qe_needed_count = self.db.report_bugzilla_qa_needed_count(data_frame)
+        self.db.report_bugzilla_qa_needed_count_insert(qe_needed_count)
+
+
+class DatabaseBugzilla(Database):
+
+    def __init__(self):
+        super().__init__()
+        self.db = Database()
+
+    def report_bugzilla_qa_needed(self, payload):
+
+        selected_columns = {
+            'bug_id': 'bugzilla_key',
+            'description': 'bugzilla_summary',
+            'modification_date': 'bugzilla_modified_at',
+            'name': 'bugzilla_tag_name',
+            'creation_date': 'bugzilla_created_at',
+            'status': 'bugzilla_tag_status',
+            'setter': 'bugzilla_tag_setter',
+            'severity': 'bugzilla_bug_severity',
+            'priority': 'bugzilla_bug_priority',
+            'bug_status': 'bugzilla_bug_status',
+            'resolution': 'bugzilla_bug_resolution'
+        }
+
+        # Select specific columns
+        df = payload[selected_columns.keys()]
+
+        # Rename columns
+        df = df.rename(columns=selected_columns)
+        return df
+
+    def report_bugzilla_qa_needed_insert(self, payload):
+        for index, row in payload.iterrows():
+            print(row)
+            report = ReportBugzillaQENeeded(
+                        bugzilla_key=row['bugzilla_key'],
+                        bugzilla_summary=row['bugzilla_summary'],
+                        buzilla_modified_at=row['bugzilla_modified_at'].date(),
```

Comment:
Does this need any date formatting, as it's using the default returned from `date()`

Reply:
let me double check... that is needed to format the entries to be added to the ddbb